### PR TITLE
Screenshot data

### PR DESCRIPTION
**Pull request by Newtsuck**
New Features:
- Can now get raw screenshot data (in PNG format) without saving a
  screenshot to the system

Improvements:
- Updated documentation, adding a short guide to delegates

### CODE REVIEW

File: docs/guides/04_Native/02_loom_native_bindings.md [hunk 1]
```diff
@@ -181,6 +181,70 @@ public native function addChild(child:MyClass, zOrder:int = 0, tag:int = 0):void
...
+Each time `pushArgument` is called, a new argument will be added to the delegate with the type of the object provided to the `pushArgument` function, for example:
+
+~~~cpp
+// stringData is a C-String
+_delegateNameDelegate.pushArgument(stringData);
+
+// numberData is an int
+_delegateNameDelegate.pushArgument(numberData);
+
+// byteArrayData is a utByteArray
```
Comment (bengarney) [inline docs/guides/04_Native/02_loom_native_bindings.md]: utByteArray *

File: docs/guides/04_Native/02_loom_native_bindings.md [hunk 2]
```diff
@@ -181,6 +181,70 @@ public native function addChild(child:MyClass, zOrder:int = 0, tag:int = 0):void
...
+.addStaticProperty("myStaticDelegate", &MyClass::getstaticDelegateNameDelegate)
+~~~
+
+**LoomScript side**
+
+Adding the native delegate to your LoomScript class is also simple!
+
+~~~{as}
+// The delegate is ready to be used!
+public native var myDelegate:NativeDelegate;
```
Comment (bengarney) [inline docs/guides/04_Native/02_loom_native_bindings.md]: You can actually do Delegate.<ArgType, ArgType, ArgType> and this is preferred. (Please test that this is right.)

Comment (Newtsuck) [inline docs/guides/04_Native/02_loom_native_bindings.md]: As far as I can tell the syntax `Delegate.<ArgType, ArgType, ArgType>` is not supported. Trying to build using this syntax fails with a "identifier expected" error, and no other code uses this format.

Comment (bengarney) [inline docs/guides/04_Native/02_loom_native_bindings.md]: I lied about the syntax: http://docs.theengine.co/loom/1.1.3452/guides/02_LoomScript/02_syntax.html#toc_16 This should work for native delegates too.

**Comment by bengarney** [comment]
What's your plan re: the Delegate syntax capability?

**Comment by Newtsuck** [comment]
I'll investigate the Delegate syntax, but the syntax I highlighted is the only syntax I have seen, can you provide a specific example?

**Comment by bengarney** [comment]
Yes:

delegate MyDelegate(arg:int);
public native var myCallback:MyDelegate;

**Comment by bengarney** [comment]
One nit, otherwise great.

**Comment by bengarney** [comment]
Great, ready to land when green.

**Pull request merged**